### PR TITLE
Increase unit test coverage

### PR DESCRIPTION
**Pull request by dchiu911**
- removed `is_absolute_path` and `isFALSE` since they aren't used (although the latter can be used [here](https://github.com/tidyverse/readxl/blob/master/R/read_excel.R#L272))
- added `mtcars`, `mtcars.txt` invalid extension files to ` tests/testthat/sheets/` for test coverage of `read_excel`
- add unit tests for `readxl_example`

### CODE REVIEW

File: tests/testthat/test-example.R [hunk 1]
```diff
@@ -0,0 +1,11 @@
+context("readxl_example")
+
+test_that("NULL path returns names of example files", {
+  example_files <- readxl_example()
+  expect_length(example_files, 10L)
```
Comment (jennybc) [inline tests/testthat/test-example.R]: Will you make this something less precise? I suggest you check that there's a match for `"datasets.xlsx"` and `"datasets.xls"`. One of those is already used below and these files likely to stick around.

File: tests/testthat/test-read-excel.R [hunk 2]
```diff
@@ -81,3 +81,28 @@ test_that("xls with exactly 65536 rows does not enter infinite loop", {
...
+  expect_error(
+    read_excel(test_sheet("mtcars.txt")),
+    "Extension is neither 'xlsx' nor 'xls': "
+  )
+  expect_error(
+    read_excel(test_sheet("mtcars")),
+    "File has no extension and doesn't seem to be xlsx or xls: "
+  )
+})
+
```
Comment (jennybc) [inline tests/testthat/test-read-excel.R]: Will you remove this? I'm reworking related functions in another PR and will have to cover this ground there.

**Comment by jennybc** [comment]
@dchiu911 The CRAN train is leaving the station very soon FYI.

**Comment by dchiu911** [comment]
@jennybc comments addressed, hopefully I made it on board! 🚆

**Comment by jennybc** [comment]
Thanks!

**Comment by jennybc** [comment]
Hey @dchiu911 I meant to squash and merge this. I'm going to get rid of this rebase. Will you open a new PR with where we ended up here?

**Comment by jennybc** [comment]
I could achieve the result myself but then you don't get the credit. I am still fussing with a few things before CRAN submission, but it's soon.

**Comment by dchiu911** [comment]
@jennybc see #458

**Pull request merged**